scipy/optimize/_linprog_util.py

```diff
@@ -597,9 +597,9 @@ def _presolve(lp, rr, tol=1e-9):
     m_eq, n = A_eq.shape
     m_ub, n = A_ub.shape
 
-    if (sps.issparse(A_eq)):
-        A_eq = A_eq.tolil()
-        A_ub = A_ub.tolil()
+    if sps.issparse(A_eq):
+        A_eq = A_eq.tocsr()
+        A_ub = A_ub.tocsr()
```

Comment:
The lack of savings in your test makes a lot of sense, because there's no LIL-native matvec implementation: it just calls `.tocsr()` under the hood.

So the big win is that by keeping everything in CSR format from the beginning, we can avoid the cost of format conversion. As for the other sparse operations, LIL is often slower than CSR, and is certainly more memory hungry.